server/core/src/irodsReServer.cpp

```diff
@@ -520,8 +520,17 @@ namespace {
             if (queue.contains_rule_id(rule_id)) {
                 return;
             }
+
             rodsLog(LOG_DEBUG, "Enqueueing rule [%s]", rule_id.data());
-            queue.enqueue_rule(rule_id);
+
+            try {
+                queue.enqueue_rule(rule_id);
+            }
+            catch (const std::bad_alloc& e) {
+                rodsLog(LOG_WARNING, "Delay queue memory limit reached. Ignoring rule [{}] for now.", rule_id.c_str());
```

Comment:
change `{}` to `%s`.

what log level should this be? perhaps `LOG_NOTICE`?

Reply:
no, i think WARNING is good for the admin - they should know they're hitting this limit - as a signal to give it more... perhaps.

course... counter argument is they set it the way they wanted to... so it should just be NOTICE.

hmmm, maybe switch on whether the setting is present in the config?   if present/defined, then it's a NOTICE.

Reply:
it might be worth incorporating something that can report the allocation attempts vs failures. that could help the admin define an appropriate memory cap on the delay queue.

that same metric could help us make a smarter delay server as well (e.g. auto adjustments to the size of the delay queue).

Reply:
LOG_DEBUG6 or LOG_DEBUG7 it is.